js/apps/system/_admin/aardvark/APP/frontend/js/views/collectionsView.js

```diff
@@ -434,20 +434,27 @@
             wfs: wfs,
             isSystem: isSystem,
             collType: collType,
-            shards: shards,
             shardKeys: shardKeys
           };
 
-          if (smartJoinAttribute !== '') {
-            tmpObj.smartJoinAttribute = smartJoinAttribute;
-          }
+          if (!isOneShardDB) {
+            if (smartJoinAttribute !== '') {
+              tmpObj.smartJoinAttribute = smartJoinAttribute;
+            }
 
-          tmpObj.distributeShardsLike = distributeShardsLike;
-          if (distributeShardsLike === '' && window.App.isCluster) {
-            // if we are in the cluster and are not using distribute shards like
-            // then we want to make use of the replication factor
-            tmpObj.replicationFactor = replicationFactor === "satellite" ? replicationFactor : Number(replicationFactor);
-            tmpObj.writeConcern = Number(writeConcern);
+            // If we are in a oneShardDB we are not allowed to set those values
+            // They are always inferred
+            if (distributeShardsLike === '') {
+              // if we are not using distribute shards like
+              // then we want to make use of the given shard information
+              tmpObj.shards = shards;
+              tmpObj.replicationFactor = replicationFactor === "satellite" ? replicationFactor : Number(replicationFactor);
+              tmpObj.writeConcern = Number(writeConcern);
+            } else {
+              // If we use distribute shards like on purpose do not add other
+              // sharding information. All of it will be deferred.
+              tmpObj.distributeShardsLike = distributeShardsLike;
+            }
```

Comment:
It seems that this code is now executed for any !oneShardDB condition. Does it mean it is also executed in single server now? I think in single server we don't want to set any of these attributes. The previous code only executed the code in question when `window.App.isCluster` was set to true.

Reply:
Yes I did this on purpose.
On SingleServer the attributes are set and retains, but they will not have any effect.
This helps with dump/restore between single-server and clusters.

Happy to revert this, but in my opinion we should make the differences between single-server and cluster in terms of Meta-Data smaller.